x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/allowed_values/helpers.tsx

```diff
@@ -5,40 +5,38 @@
  * 2.0.
  */
 
-import type { EcsMetadata, UnallowedValueRequestItem } from '../../types';
+import type { EcsFlatTyped } from '../../constants';
+import type { EcsFieldMetadata, UnallowedValueRequestItem } from '../../types';
 
 export const hasAllowedValues = ({
   ecsMetadata,
   fieldName,
 }: {
-  ecsMetadata: Record<string, EcsMetadata> | null;
+  ecsMetadata: EcsFlatTyped;
   fieldName: string;
-}): boolean =>
-  ecsMetadata != null ? (ecsMetadata[fieldName]?.allowed_values?.length ?? 0) > 0 : false;
+}): boolean => (ecsMetadata[fieldName]?.allowed_values?.length ?? 0) > 0;
 
-export const getValidValues = (field: EcsMetadata | undefined): string[] =>
-  field?.allowed_values?.flatMap(({ name }) => (name != null ? name : [])) ?? [];
+export const getValidValues = (field?: EcsFieldMetadata): string[] =>
+  field?.allowed_values?.flatMap(({ name }) => name) ?? [];
 
 export const getUnallowedValueRequestItems = ({
   ecsMetadata,
   indexName,
 }: {
-  ecsMetadata: Record<string, EcsMetadata> | null;
+  ecsMetadata: EcsFlatTyped;
   indexName: string;
 }): UnallowedValueRequestItem[] =>
-  ecsMetadata != null
```

Comment:
For reviewers:

we remove this check because it checks an impossible state. `ecsMetadata` represents `EcsFlat` object which is imported statically so it can never be null.